lib/engine/game/g_18_ms.rb

```diff
@@ -12,18 +12,61 @@ class G18MS < Base
       GAME_LOCATION = 'Mississippi, USA'
       GAME_DESIGNER = 'Mark Derrick'
       GAME_PUBLISHER = Publisher::INFO[:all_aboard_games]
-      GAME_END_CHECK = { final_or_set: 10 }.freeze
+      GAME_INFO_URL = 'https://github.com/tobymao/18xx/wiki/18MS'
+
+      # Game ends after 5 * 2 ORs
+      GAME_END_CHECK = { final_or_set: 5 }.freeze
 
       HOME_TOKEN_TIMING = :operating_round
 
+      COMPANY_1_AND_2 = %w[AGS BS].freeze
+
+      STATUS_TEXT = Base::STATUS_TEXT.merge(
+        'can_buy_companies_operation_round_one' =>
+          ['Can Buy Companies OR 1', 'Corporations can buy AGS/BS companies for face value in OR 1'],
+      ).freeze
+
       #      def init_round
       #        Round::G18MS::Draft.new(@players.reverse, game: self)
       #      end
 
       include CompanyPrice50To150Percent
 
       def setup
-        setup_company_price_50_to_150_percent
+        @previously_floated_corporations = []
+      end
+
+      def purchasable_companies(entity = nil)
+        entity ||= current_entity
+        # Only companies owned by the president may be bought
+        companies = super.reject do |c|
+          entity.corporation? && entity.player == c.player
```

Comment:
do you want this, or do you want owned_by?(entity)

Reply:
Yes, but it was reversed check. Have corrected that now, and using owned_by?